src/py42/_internal/clients/alerts.py

```diff
@@ -73,19 +79,45 @@ def get_all_rules(self, sort_key=u"CreatedAt", sort_direction=u"DESC"):
             self._get_alert_rules,
             u"ruleMetadata",
             tenant_id=tenant_id,
+            groups=None,
             sort_key=sort_key,
             sort_direction=sort_direction,
         )
 
-    def get_rules_by_name(self, rule_name):
-        rule_pages = self.get_all_rules()
-        matched_rules = []
-        for rule_page in rule_pages:
-            rules = rule_page[u"ruleMetadata"]
-            for rule in rules:
-                if rule_name.lower() == rule[u"name"].lower():
-                    matched_rules.append(rule)
-        return matched_rules
+    def get_all_rules_by_name(self, rule_name, sort_key=u"CreatedAt", sort_direction=u"DESC"):
```

Comment:
Basically by looking at the source code I was able to figure out that these filter clauses work. I don't think that rules are complex that we need to implement the full query system for them, at least not yet. If there is demand for that we can keep still add that and keep these functions.